lib/grpc/GrpcServer.ts

```diff
@@ -36,9 +36,12 @@ class GrpcServer {
    */
   public listen = (port: number, host: string) => {
     assert(Number.isInteger(port) && port > 1023 && port < 65536, 'port must be an integer between 1024 and 65535');
-
     try {
-      this.server.bind(`${host}:${port}`, grpc.ServerCredentials.createInsecure());
+      const bindCode = this.server.bind(`${host}:${port}`, grpc.ServerCredentials.createInsecure());
+      if (bindCode !== port) {
+        this.logger.error(`gRPC Failed to listen on port: ${port}`);
+        throw `gRPC couldn't bind on port: ${port}`;
```

Comment:
Small remarks:
* better not to throw a string.
* you can avoid the local scope logging, since you're not adding any further information
* you can drop the `gRPC` prefix in the local scope logging, since you're already in that context.